### ISSUE TITLE
networkd does not support iproute2 onlink

### ISSUE THREAD

**Issue by guerby** [systemd/systemd#1283]
Hi,

When you have a gateway whose IP is not on the subnet iproute2 has an option "onlink" to says the gateway is on the interface, eg:

```
# ip route add default via 1.2.3.4 dev eth0 onlink
# ip route
default via 1.2.3.4 dev eth0  proto static onlink
#
```

This feature is supported by OpenWRT for example because it's quite useful in some routing setups.

As a workaround you can try to emulate this feature with two routes :

```
# ip route add 1.2.3.4/32 dev eth0
# ip route add default via 1.2.3.4
```

But I couldn't make it work with systemd [Route] sections, may be an ordering or check issue.

We had to remove systemd network unit and write a custom script unit because of this limitation.

**Comment by guerby** [comment]
Looks like there's a workaround:

http://serverfault.com/questions/667319/systemd-networkd-and-direct-routes

 [Match]
 Name=enp0s3

[Address] 
Address=192.168.0.2
Peer=192.168.0.1/32

[Network]
Gateway=192.168.0.1

**Comment by ssahani** [comment]
we need to close this as PR is merged.

**Comment by poettering** [comment]
Closing, as this has been implemented in #5734 now.
